course_discovery/apps/course_metadata/data_loaders/api.py

```diff
@@ -899,3 +899,99 @@ def _update_program_banner_image(self, body, program):
             program.save()
         else:
             logger.exception('Loading the banner image %s for program %s failed', image_url, program.title)
+
+
+class WordPressApiDataLoader(AbstractDataLoader):
+    """
+    Sync courses data from WordPress.
+    """
+
+    def ingest(self):
+        """
+        Load courses data from the WordPress.
+        """
+        logger.info('Refreshing Courses Data from WordPress %s...', self.partner.marketing_site_api_url)
+        initial_page = 1
+        response = self._make_request(initial_page)
+        count = response['pagination']['count']
+        pages = response['pagination']['num_pages']
+        self._process_response(response)
+
+        pagerange = range(initial_page + 1, pages + 1)
+        logger.info('Looping to request all %d WordPress pages...', pages)
+
+        with concurrent.futures.ThreadPoolExecutor(max_workers=self.max_workers) as executor:
+            if self.is_threadsafe:
+                for page in pagerange:
+                    time.sleep(30)
```

Comment:
It looks like we are able to use most of the code from existing loaders. 👍 
<img width="883" alt="Screen Shot 2021-02-03 at 7 18 51 PM" src="https://user-images.githubusercontent.com/5072991/106759827-ffcf6500-6654-11eb-81bc-c81b3b1d8867.png">

Reply:
Yes, lucky that snippet works for us :D